scipy/optimize/_differentiable_functions.py

```diff
@@ -407,27 +407,45 @@ class LinearVectorFunction(object):
     A is m-by-n matrix. The Jacobian is constant and equals to A. The Hessian
     is identically zero and it is returned as a csr matrix.
     """
-    def __init__(self, A, sparse_jacobian):
+    def __init__(self, A, x0, sparse_jacobian):
         if sparse_jacobian or sparse_jacobian is None and sps.issparse(A):
-            self.A = sps.csr_matrix(A)
+            self.J = sps.csr_matrix(A)
             self.sparse_jacobian = True
         elif sps.issparse(A):
-            self.A = A.toarray()
+            self.J = A.toarray()
             self.sparse_jacobian = False
         else:
-            self.A = np.atleast_2d(A)
+            self.J = np.atleast_2d(A)
             self.sparse_jacobian = False
 
-        self.m, self.n = self.A.shape
+        self.m, self.n = self.J.shape
+
+        self.x = np.atleast_1d(x0).astype(float)
+        self.f = self.J.dot(self.x)
+        self.f_updated = True
+
+        self.v = np.zeros(self.m, dtype=float)
         self.H = sps.csr_matrix((self.n, self.n))
 
+    def change_x(self, x):
+        if not np.array_equal(x, self.x):
```

Comment:
Hi Nikolay,

I know this is quite minor, but I think in order to make the implementation more similar to `VectorFunction`. It would be better if we define:
```
def update_x(self, x):
    self.x = x
    self.f_updated = False
```

and:

```
def fun(self, x):
    if not np.array_equal(x, self.x):
        self.update(x)
...
def jac(self, x):
    if not np.array_equal(x, self.x):
        self.update(x)
...
def hess(self, x):
    if not np.array_equal(x, self.x):
        self.update(x)
...
```